lib/handleRequest.js

```diff
@@ -133,21 +136,31 @@ function preHandlerCallback (err, request, reply) {
 
   let result
 
+  const store = { request, reply }
+
+  channels.start.publish(store)
+
   try {
     result = request[kRouteContext].handler(request, reply)
   } catch (err) {
+    store.error = err
+    channels.error.publish(store)
+
     reply[kReplyIsError] = true
     reply.send(err)
     return
   }
 
   if (result !== undefined) {
+    store.result = result
```

Comment:
The properly follow the tracing channel spec, the `store.result` value should be set only for the sync branch below and the async branch through `wrapThenable` should set `store.result` to the `payload` object rather than storing that in `store.payload`.

Reply:
Sounds good. It did feel like I was doing something wrong by swapping the meaning of `.result` depending on if it were an async or sync handler.